credentials/alts/internal/conn/record_test.go

```diff
@@ -105,6 +108,94 @@ func newConnPair(rp string, clientProtected []byte, serverProtected []byte) (cli
 	return clientConn, serverConn
 }
 
+// newTCPConnPair returns a pair of conns backed by TCP over loopback.
+func newTCPConnPair(rp string, clientProtected []byte, serverProtected []byte) (*conn, *conn, error) {
```

Comment:
Nit: Please consider replacing `rp` with `recordProtocol` as really short variable names make more sense where the scope of the variable is much smaller. This seems to be a reasonable large function, and `rp` is used way down below.

Also, `clientProtected []byte, serverProtected []byte` could be shortened as `clientProtected, serverProtected []byte`